src/app/shared/object-list/sidebar-search-list-element/item-types/publication/publication-sidebar-search-list-element.component.ts

```diff
@@ -0,0 +1,23 @@
+import { listableObjectComponent } from '../../../../object-collection/shared/listable-object/listable-object.decorator';
+import { ViewMode } from '../../../../../core/shared/view-mode.model';
+import { Component } from '@angular/core';
+import { Context } from '../../../../../core/shared/context.model';
+import { ItemSearchResult } from '../../../../object-collection/shared/item-search-result.model';
+import { Item } from '../../../../../core/shared/item.model';
+import { SidebarSearchListElementComponent } from '../../sidebar-search-list-element.component';
+
+@listableObjectComponent('PublicationSearchResult', ViewMode.ListElement, Context.SideBarSearchModal)
+@listableObjectComponent('PublicationSearchResult', ViewMode.ListElement, Context.SideBarSearchModalCurrent)
+@listableObjectComponent(ItemSearchResult, ViewMode.ListElement, Context.SideBarSearchModal)
+@listableObjectComponent(ItemSearchResult, ViewMode.ListElement, Context.SideBarSearchModalCurrent)
+@Component({
+  selector: 'ds-publication-sidebar-search-list-element',
+  templateUrl: '../../sidebar-search-list-element.component.html'
+})
+/**
+ * Component displaying a list element for a {@link ItemSearchResult} of type "Publication" within the context of
+ * a sidebar search modal
+ */
+export class PublicationSidebarSearchListElementComponent extends SidebarSearchListElementComponent<ItemSearchResult, Item> {
```

Comment:
I'm surprised to see this named `Publication` (which is an type of Entity) instead of just being a generic `ItemSidebarSearchListElement`.  Logically, many more people are going to use generic Items than Publication Entities.  So, even if this same file can be used for *both* types, we probably should think about naming it `ItemSidebarSearchListElement`.

That said, as I noticed this, I looked around in other areas of the code, and we seem to have developed an "antipattern" of having components starting with `Publication*` even if they also are used for generic Items....another example is the `PublicationComponent` itself: https://github.com/DSpace/dspace-angular/blob/main/src/app/%2Bitem-page/simple/item-types/publication/publication.component.ts  (which seems to have a side effect of displaying "Publication:" before every Item title.

So, we don't necessarily need to fix this in this PR.  However, I'd argue we need to change the names of these components to start with `Item` or `GenericItem`.  I can open this as a separate ticket though, as I suspect this is better done separately.

Reply:
I've logged the above comment in #936